src/commonMain/kotlin/me/alllex/parsus/token/RegexToken.kt

```diff
@@ -56,3 +56,11 @@ fun GrammarContext.regexToken(
     ignored: Boolean = false
 ): RegexToken = RegexToken(regex, name, ignored)
 
+//internal expect fun Regex.toMatcher(): RegexMatcher
+//
+//internal interface RegexMatcher {
+//    fun reset(input: CharSequence)
+//    fun region(start: Int, end: Int)
+//    fun find(): Boolean
+//    fun end(): Int
+//}
```

Comment:
This code (and the `actual` in jvmMain) is another potential way of handling `Matcher` being JVM only. I went with refactoring the code above so that Regex would work, because Regex is already multiplatform.

I've left this code in as a demo, but I think it should be removed.